warp10/src/main/java/io/warp10/script/functions/LIMIT.java

```diff
@@ -18,39 +18,42 @@
 
 import io.warp10.script.NamedWarpScriptFunction;
 import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.warp.sdk.Capabilities;
 import io.warp10.script.WarpScriptException;
 import io.warp10.script.WarpScriptStack;
 
 /**
  * Configure the fetch limit of the stack
  */
 public class LIMIT extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
   public LIMIT(String name) {
     super(name);
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
-    
-    if (!stack.isAuthenticated()) {
-      throw new WarpScriptException(getName() + " requires the stack to be authenticated.");
+
+    if (null == Capabilities.get(stack, WarpScriptStack.CAPABILITY_LIMITS) && null == Capabilities.get(stack, WarpScriptStack.CAPABILITY_LIMIT)) {
+      throw new WarpScriptException(getName() + " missing capability.");
     }
-    
+
     Object top = stack.pop();
-    
+
     if (!(top instanceof Long)) {
-      throw new WarpScriptException(getName() + " expects a numeric (long) limit.");
+      throw new WarpScriptException(getName() + " expects a numeric (LONG) limit.");
     }
-    
+
     long limit = ((Number) top).longValue();
 
-    if (limit > (long) stack.getAttribute(WarpScriptStack.ATTRIBUTE_FETCH_LIMIT_HARD)) {
-      throw new WarpScriptException(getName() + " cannot extend limit past " + stack.getAttribute(WarpScriptStack.ATTRIBUTE_FETCH_LIMIT_HARD));
+    Long max = Capabilities.getLong(stack, WarpScriptStack.CAPABILITY_LIMIT, (long) stack.getAttribute(WarpScriptStack.ATTRIBUTE_FETCH_LIMIT_HARD));
+
+    if (limit > max) {
+      throw new WarpScriptException(getName() + " cannot extend limit past " + max);
```

Comment:
also indicate WarpScriptStack.ATTRIBUTE_FETCH_LIMIT_HARD in the error message to guide user ? or "past beyond the configuration hard limit " ?